gnmi_subscribe/gnmi_subscribe.go

```diff
@@ -176,11 +231,55 @@ func assembleSubscriptions(paths []*pb.Path) ([]*pb.Subscription, error) {
 		subscription := &pb.Subscription{
 			Path:              path,
 			Mode:              subscriptionMode,
-			SampleInterval:    *sampleInterval,
+			SampleInterval:    sampleInterval,
 			SuppressRedundant: *suppressRedundant,
 			HeartbeatInterval: *heartbeatInterval,
 		}
 		subscriptions = append(subscriptions, subscription)
 	}
 	return subscriptions, nil
 }
+
+func subscriptionMode(subscriptionPoll, subscriptionOnce bool) (gnmi.SubscriptionList_Mode, error) {
+	switch {
+	case subscriptionPoll && subscriptionOnce:
+		return UNKNOWN, errors.New("only one of -once and -poll can be set")
```

Comment:
just return 0, it does not matter what this value is as there is a non nil error.